### DISCUSSION TITLE
SMB Volume - Operation not permitted

### DISCUSSION THREAD

**Discussion by RodCSouza** [alexjustesen/speedtest-tracker#1123] [Q&A]
Hello guys,

I'm trying to execute the image, using a docker compose file and using volume from a SMB shared folder, but I'm getting the "operation not permitted" error, in the changing permissions step.
This is the docker compose file content:
```dockerfile
version: "3.3"
services:
    speedtest-tracker:
        container_name: speedtest-tracker
        ports:
            - "8380:80"
            - "8343:443"
        environment:
            - PUID=1000
            - PGID=1000
            - DB_CONNECTION=sqlite
            - DB_DATABASE=/config/database.sqlite
        volumes:
            - speedtest-app:/config
            - "/path/to/directory/web:/etc/ssl/web"
        image: "ghcr.io/alexjustesen/speedtest-tracker:latest"
        restart: unless-stopped
volumes:
  speedtest-app:
    driver: local
    driver_opts:
      type: "nfs"
      o: "addr=xx.xx.xx.xx"
      device: ":/bla/blablabla/nono/speedtracker/config"

``` 
The SMB folder is from a TrueNAS server, where the user permissions are already set, meaning that would not be necessary to run this "changing permissions" step, at least from my simplest understanding, or if it is possible to determine the user to be used on the container creation, meaning that I can set the same as the one within the permission in place.

Could you please advise?

Thanks

**Comment by alexjustesen** [comment]
Did you run `id {username}` to get your PUID and PGID? Most permission issues can be fixed by running that on the user that runs your docker containers.
